database/ticket.go

```diff
@@ -114,23 +127,114 @@ func (vdb *VspDatabase) InsertNewTicket(ticket Ticket) error {
 			return err
 		}
 
-		ticketBytes, err := json.Marshal(ticket)
+		err = putTicketInBucket(newTicketBkt, ticket)
 		if err != nil {
-			return fmt.Errorf("could not marshal ticket: %w", err)
+			return fmt.Errorf("putting ticket in bucket failed: %w", err)
 		}
 
-		return ticketBkt.Put(hashBytes, ticketBytes)
+		return nil
 	})
 }
 
-func (vdb *VspDatabase) DeleteTicket(ticket Ticket) error {
+func putTicketInBucket(bkt *bolt.Bucket, ticket *Ticket) error {
+	var err error
+	if err = bkt.Put(HashK, []byte(ticket.Hash)); err != nil {
+		return err
```

Comment:
Suggest wrapping these put errors to provide more context. A description for the function would be great too.

Reply:
I've added a comment for the func.

I am reluctant to add additional error handling code here because it is already quite verbose/repetitive, and the chance of an error happening on these lines is negligible. Looking at the comments for `Put()`...

> Returns an error if the bucket was created from a read-only transaction,

Our code is always creating an Update transaction

> if the key is blank, if the key is too large,

Keys are hard-coded constant values which are neither blank nor too large (limit is 32768 chars)

> or if the value is too large.

Values are always of fixed or known length, and besides that, the maximum limit is 2,147,483,646

Reply:
gotcha, I initially wanted a more descriptive error considering its a number of clustered put calls, but this should be fine since in case of an error we can find the put call that erred with the line number.